### PR TITLE
Update cmake to 3.19.1

### PR DESCRIPTION
**Pull request by traversaro**
Bump CMake to version 3.19.1 .

3.19.1 Release Notes : https://discourse.cmake.org/t/cmake-3-19-1-available-for-download/2228
3.19.0 Release Notes : https://discourse.cmake.org/t/cmake-3-19-0-available-for-download/2198

Fix https://github.com/conda-forge/ruby-feedstock/issues/43 .

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by scopatz** [comment]
This seems to fail on win and osx.

**Comment by traversaro** [comment]
> 
> 
> This seems to fail on win and osx.

The Windows failure seems the good old "the user forgot to enable openssl" but indeed it seems that it was not enabled before, probably there was some auto-detected options that was enabling it before. I can look into enabling openssl explicitly just to be sure. Good old `osx-amd64`, what is failing is the new `osx-arm64`, but I can't fully understand the error.

**Comment by scopatz** [comment]
Yeah, I am not sure about the osx-amd64 issues.  I would be ok allowing that platform as a failure since it so new.  But we should check with other @conda-forge/cmake maintainers

**Comment by traversaro** [comment]
Thanks @isuruf for fixing osx-arm64!

**Comment by traversaro** [comment]
The Windows failure appears due to the fact that CMAKE_USE_WINSSL option was renamed to CMAKE_USE_SCHANNEL, and moved from ON by default to OFF by default (not sure if it was intentional).

File: recipe/build.sh [hunk 1]
```diff
@@ -6,14 +6,19 @@ CMAKE_ARGS="$CMAKE_ARGS -DCURSES_INCLUDE_PATH=${PREFIX}/include -DBUILD_CursesDi
 CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_PREFIX_PATH=${PREFIX}"
 
 if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
+   if [[ "$target_platform" == osx-* ]] && [[ "$MACOSX_DEPLOYMENT_TARGET" == 11.* || "$MACOSX_DEPLOYMENT_TARGET" == "10.15" ]]; then
```
Comment (scopatz) [inline recipe/build.sh:9]: @isuruf - do you want to add yourself as a maintainer here too?

Comment (isuruf) [reply recipe/build.sh:9]: No, I don't. cc @conda-forge/core

Comment (scopatz) [reply recipe/build.sh:9]: Isn't it more correct to ask @conda-forge/cmake?

**Pull request merged**